s3_storage_provider.py

```diff
@@ -84,6 +85,11 @@ def __init__(self, hs, config):
         if "session_token" in config:
             self.api_kwargs["aws_session_token"] = config["session_token"]
 
+        self.api_kwargs["config"] = Config(
+            response_checksum_validation=config.get("response_checksum_validation", 'when_required'),
+            request_checksum_calculation=config.get("request_checksum_calculation", 'when_required')
```

Comment:
Avoid mixing `'` and `"` string boundary characters.

```suggestion
            response_checksum_validation=config.get("response_checksum_validation", "when_required"),
            request_checksum_calculation=config.get("request_checksum_calculation", "when_required")
```